### PR TITLE
Empty package for `package:cli_config`

### PR DESCRIPTION
**Pull request by dcharkes**
Sets up the directory structure for `package:cli_config`.

The package contains no Dart code yet, this will be added later.

### CODE REVIEW

File: CODEOWNERS [hunk 1]
```diff
@@ -2,4 +2,5 @@
 # for code within the repository
 # https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners
 
+/pkgs/cli_config/ @dcharkes @HosseinYousefi
```
Comment (devoncarew) [inline CODEOWNERS:5]: Having an entry here is your choice; we don't generally use the CODEOWNERS file. Elias wanted to use one (or experiment with one?).

The benefit here is that you'll automatically be put as a reviewer for every PR that touches a file matching the glob above.

File: README.md [hunk 2]
```diff
@@ -14,6 +14,7 @@ don't naturally belong to other topic monorepos (like
 
 | Package | Description | Version |
 | --- | --- | --- |
+| [cli_config](pkgs/cli_config/) | A package to take config values from configuration files, CLI arguments, and environment variables. |
```
Comment (devoncarew) [inline README.md:17]: nit: you can end the markdown table here w/ ` | |` (an empty column for the pub badge)

File: pkgs/cli_config/CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,3 @@
+## 0.1.0
```
Comment (devoncarew) [inline pkgs/cli_config/CHANGELOG.md:1]: I'd change to `0.1.0-dev` until we're ready to publish the first version.

File: pkgs/cli_config/README.md [hunk 4]
```diff
@@ -0,0 +1 @@
+A library to take config values from configuration files, CLI arguments, and environment variables.
```
Comment (devoncarew) [inline pkgs/cli_config/README.md:1]: We should expand on this before publishing (not necessarily for this PR).

File: pkgs/cli_config/analysis_options.yaml [hunk 5]
```diff
@@ -0,0 +1,22 @@
+include: package:lints/recommended.yaml
```
Comment (devoncarew) [inline pkgs/cli_config/analysis_options.yaml:1]: 👍

File: pkgs/cli_config/pubspec.yaml [hunk 6]
```diff
@@ -0,0 +1,16 @@
+name: cli_config
+description: A library to take config values from configuration files, CLI arguments, and environment variables.
+version: 0.1.0
```
Comment (devoncarew) [inline pkgs/cli_config/pubspec.yaml:3]: ==> `0.1.0-dev`

File: pkgs/cli_config/pubspec.yaml [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+
+repository: https://github.com/dcharkes/config
```
Comment (devoncarew) [inline pkgs/cli_config/pubspec.yaml:5]: ==> `https://github.com/dart-lang/tools/tree/main/pkgs/cli_config`

**Review by devoncarew — APPROVED** [review]
lgtm w/ the indicated changes

File: CODEOWNERS (hunk 1 above)
Comment (dcharkes) [reply CODEOWNERS:5]: Check, removed again.

**Pull request merged**